domain/service/servicedetails_store.go

```diff
@@ -73,6 +62,32 @@ func (s *storeImpl) GetAllServiceDetails(ctx datastore.Context, since time.Durat
 		}
 
 		parentMap[d.ParentServiceID] = struct{}{}
+
+		if r != nil && !r.MatchString(d.Name) {
+			continue
+		}
+
+		if query.Since > 0 && d.UpdatedAt.Before(since) {
+			continue
+		}
+
+		if query.Tenants && d.ParentServiceID != "" {
```

Comment:
I started with that approach but encountered an issue with the name parameter. In the v1 endpoint, the name specified can be a regex and it used Go's regex library to evaluate.  The regex in elastic is evaluated differently so I was getting different results between the two for the same name.  I wanted to keep the same functionality as the v1 endpoint so I have to evaluate the regex server side in Go.  Since I had to do it server side and go through all the results anyway, I just added the other checks as well to make it more consistent.